Client/Frontend/Browser/Favicons/FaviconFetcher.swift

```diff
@@ -438,24 +445,28 @@ class FaviconFetcher {
                             score += 1
                         }
                     }
-                    for x in 0..<Int(icon.size.width) {
+                    for x in 0..<Int(iconSize.width) {
                         updateScore(x: x, y: 0)
                     }
-                    for x in 0..<Int(icon.size.width) {
-                        updateScore(x: x, y: Int(icon.size.height))
+                    for x in 0..<Int(iconSize.width) {
+                        updateScore(x: x, y: Int(iconSize.height))
                     }
                     // We've already scanned the first and last pixel during
                     // top/bottom pass
-                    for y in 1..<Int(icon.size.height)-1 {
+                    for y in 1..<Int(iconSize.height)-1 {
```

Comment:
Will this crash if `iconSize .height == 1`? 
earlier in this method we check for non zero value, and with subtracting 1 we can get 1..<0 range which can crash

Reply:
Good point but I don't think this is possible, as `iconSize` will be the size of the scaled context regardless (48*scale)